www/include/views/componentTemplates/DB-Func.php

```diff
@@ -39,54 +39,42 @@
-        $gsvs = $form->getSubmitValues();
+        $formValues = $form->getSubmitValues();
     }
-    $sql = "SELECT compo_id FROM giv_components_template WHERE ";
-    $sql .= "name = '" . $gsvs["name"] . "' ";
-    if ($gsvs["host_id"] != null) {
-        list($host_id, $service_id) = explode('-', $gsvs["host_id"]);
-        $sql .= "AND host_id = '" . $host_id . "' AND service_id = '" . $service_id . "'";
-    } else {
-        $sql .= "AND host_id IS NULL  AND service_id IS NULL";
-    }
-    $dbResult = $pearDB->query($sql);
-    $compo = $dbResult->fetch();
-    #Modif case
-    if ($dbResult->rowCount() >= 1 && $compo["compo_id"] == $gsvs["compo_id"]) {
-        return true;
-        #Duplicate entry
-    } elseif ($dbResult->rowCount() >= 1 && $compo["compo_id"] != $gsvs["compo_id"]) {
-        return false;
-    } else {
-        return true;
+    $query = 'SELECT compo_id FROM giv_components_template WHERE name = :name';
+    if (!empty($formValues['host_id'])) {
+        if (preg_match('/([0-9]+)-([0-9]+)/', $formValues['host_id'], $matches)) {
+            $formValues['host_id'] = (int) $matches[1];
+            $formValues['service_id'] = (int) $matches[2];
+        } else {
+            throw new \InvalidArgumentException('chartId must be a combination of integers');
+        }
     }
-}
 
-function DsHsrTestExistence($name = null)
-{
-    global $pearDB, $pearDBO, $form;
-    $gsvs = null;
-    if (isset($form)) {
-        $gsvs = $form->getSubmitValues();
+    $query .= (empty($formValues['host_id']) && empty($formValues['service_id']))
+        ? ' AND host_id IS NULL  AND service_id IS NULL'
+        : ' AND host_id = :host_id AND service_id = :service_id';
+
+    if (!empty($formValues)) {
+        $bindParams = sanitizeFormComponentTemplatesParameters($formValues);
     }
-    $sql = "SELECT compo_id FROM giv_components_template WHERE ";
-    $sql .= "ds_name = '" . $gsvs["ds_name"] . "' ";
-    if ($gsvs["host_id"] != null) {
-        list($host_id, $service_id) = explode('-', $gsvs["host_id"]);
-        $sql .= "AND host_id = '" . $host_id . "' AND service_id = '" . $service_id . "'";
-    } else {
-        $sql .= "AND host_id IS NULL  AND service_id IS NULL";
+
+    $stmt = $pearDB->prepare($query);
+
+    $stmt->bindValue(':name', $name, \PDO::PARAM_STR);
+
+    foreach ($bindParams as $token => list($paramType, $value)) {
+            $stmt->bindValue($token, $value, $paramType);
     }
-    $dbResult = $pearDB->query($sql);
-    $compo = $dbResult->fetch();
-    #Modif case
-    if ($dbResult->rowCount() >= 1 && $compo["compo_id"] == $gsvs["compo_id"]) {
+    $stmt->execute();
+    $compo = $stmt->fetch();
+    if ($stmt->rowCount() >= 1 && $compo['compo_id'] == $formValues['compo_id']) {
         return true;
-    } #Duplicate entry
-    elseif ($dbResult->rowCount() >= 1 && $compo["compo_id"] != $gsvs["compo_id"]) {
+    } elseif ($stmt->rowCount() >= 1 && $compo['compo_id'] != $formValues['compo_id']) {
```

Comment:
double equals are needed ? If it's possible could you replace it by triple equals?